streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/bolt/CustomProcessorBolt.java

```diff
@@ -203,30 +170,35 @@ public void prepare (Map stormConf, TopologyContext context, OutputCollector col
     @Override
     protected void process (Tuple input, StreamlineEvent event) {
         try {
-            Map<String, Object> mappedEvent = new HashMap<>();
-            if (!inputSchemaMap.containsKey(input.getSourceStreamId()))
-                throw new RuntimeException("Received an event on an input stream that does not have mapping for input schema fields");
-            // Create a new mapped event based on mapping to pass it to CP implementation
-            for (Map.Entry<String, String> entry: inputSchemaMap.get(input.getSourceStreamId()).entrySet()) {
-                if (event.get(entry.getValue()) != null) {
-                    mappedEvent.put(entry.getKey(), event.get(entry.getValue()));
+            StreamlineEvent toProcess = null;
+            if (inputSchemaMap == null || inputSchemaMap.isEmpty() || !inputSchemaMap.containsKey(input.getSourceStreamId())) {
+                toProcess = event;
+            } else {
+                // Create a new mapped event based on mapping to pass it to CP implementation
+                Map<String, Object> mappedEvent = new HashMap<>();
+                for (Map.Entry<String, String> entry: inputSchemaMap.get(input.getSourceStreamId()).entrySet()) {
+                    if (event.get(entry.getValue()) != null) {
+                        mappedEvent.put(entry.getKey(), event.get(entry.getValue()));
+                    }
                 }
+                toProcess = new StreamlineEventImpl(mappedEvent, event.getDataSourceId(), event.getId(), event.getHeader(), input.getSourceStreamId(), event
+                        .getAuxiliaryFieldsAndValues());
             }
-            List<StreamlineEvent> results = customProcessorRuntime.process(new StreamlineEventImpl(mappedEvent, event.getDataSourceId(), event
-                    .getId(), event.getHeader(), input.getSourceStreamId(), event.getAuxiliaryFieldsAndValues()));
+            List<StreamlineEvent> results = customProcessorRuntime.process(toProcess);
             if (results != null) {
+                Schema schema = outputSchema.values().iterator().next();
```

Comment:
1. Can there be only one output stream and schema? Is that going to be addressed separately?
2. Isn;t the output schema going to be optional as a part of this PR? if so this may throw an Exception.

Reply:
Yes, its only one output schema and stream for now. It can be addressed separately based on the need for that use case. I did not plan to address it immediately.
Output schema is optional for registering the CP. This is the output schema at the component level in a SAM app. UI will force the user to create an output schema and hence this will have exactly one entry.